types/errors/errors.go

```diff
@@ -394,3 +398,24 @@ type causer interface {
 type unpacker interface {
 	Unpack() []error
 }
+
+func GRPCWrap(err error, c codes.Code, msg string) error {
+	if err == nil {
+		return nil
+	}
+	st := status.New(c, msg)
+	var sdkErr *Error
+	if errors.As(err, &sdkErr) {
+		errorInfo := &errdetails.ErrorInfo{
+			Reason:   sdkErr.Error(),
+			Metadata: map[string]string{"Codespace": sdkErr.Codespace(), "ABCICode": fmt.Sprintf("%d", sdkErr.ABCICode())},
+		}
+		var withDetailsErr error
+		st, withDetailsErr = st.WithDetails(errorInfo)
+		if withDetailsErr != nil {
+			return withDetailsErr
+		}
+	}
```

Comment:
I'm still not sure that we should return the errors found calling `WithDetails`, isn't going to mask the real error that we want to wrap?

```suggestion
		if withDetailsSt, withDetailsErr := st.WithDetails(errorInfo); withDetailsErr == nil {
			st = withDetailsSt
		}
	}
```

Reply:
Yeah, I agree but if we not return the withDetailsErr It will be hard to  identify the cause of missing error codes. In addition I think that `withDetailsErr` it's rare and it will be triggered only during implementation phase, not in prod environment, so it will be immediatly fixed.

Reply:
It's just to choose the lesser of two evils

Reply:
I agree there's no good solution, with both we lose context, I would merge the two errors then
```go
return fmt.Errorf("%w (failed to add status details: %v)", err, withDetailsErr)
```